test/integration/testpaths/testpath_hypervisor_limit.py

```diff
@@ -0,0 +1,198 @@
+        cls.host = Host(hostList[0].__dict__)
+        Host.update(cls.apiclient, id=cls.host.id, hosttags="host1")
+        try:
+            cls.skiptest = False
+            if cls.hypervisor.lower() not in ['xenserver']:
+                cls.skiptest = True
+                return
+
+            # Create an account
+            cls.account = Account.create(
+                cls.apiclient,
+                cls.testdata["account"],
+                domainid=cls.domain.id
+            )
+
+            # Create user api client of the account
+            cls.userapiclient = testClient.getUserApiClient(
+                UserName=cls.account.name,
+                DomainName=cls.account.domain
+            )
+            # Create Service offering
+            cls.service_offering = ServiceOffering.create(
+                cls.apiclient,
+                cls.testdata["service_offering"],
+                hosttags="host1"
+            )
+
+            cls._cleanup = [
+                cls.account,
+                cls.service_offering,
+            ]
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.cleanup = []
+        if self.skiptest:
+            self.skipTest("This test is to be checked on xenserver \
+                    only  Hence, skip for %s" % self.hypervisor)
+
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+
+    def tearDown(self):
+        try:
+
+            cmd = updateHypervisorCapabilities.updateHypervisorCapabilitiesCmd()
+            cmd.id = self.hostCapId
+            cmd.maxguestslimit = self.originalLimit
+            self.apiclient.updateHypervisorCapabilities(cmd)
+
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["advanced", "basic"], required_hardware="false")
+    def test_check_hypervisor_max_limit_effect(self):
+        """ Test hypervisor max limits effect
+
+        # 1. Read exsiting count of VM's on the host 
+                and modify maxguestlimit of the host by exsiting count + 4
+        # 2. Deploy a VM
+        # 2. Try to deploy another vm
+        # 3. Verify that second VM
+                deployment fails (2 SSVMs 1 VR VM and 1 deployed VM)
+        """
+
+        # Step 1
+        listVm = list_virtual_machines(self.apiclient,
+                                       hostid=self.host.id)
+        vmCount = len(listVm)
+        # Increament vm count by 4(2 SSVMs 1 VR VM and 1 deployed VM)
+        newValue = vmCount + 4
+
```

Comment:
listVirtualMachines API does not list 2 SSVMs and 1 VR it only list the User deployed VM's hence I am adding 4 to vmCount.